Refactor lazyfree slightly

ShooterIT opened the pull request:
- Create `lazyfreeGeneric` macro, in `lazyfree.c`, lazy free just is to call free function, then decrease lazyfree_objects and increase lazyfreed_objects by specific size. And most functions have similar code, we can use this macro to replace them.
- Many `lazyfree` function is to free `dict`, `list`, `rax`, so i add  corresponding  macros
- I think extra `Free` word is not necessary, so rename `lazyfreeFreeObject` to `lazyfreeObject`, `lazyfreeFreeDatabase`  to `lazyfreeDatabase`
- In `emptyDbAsync`, we should count `expires` dict size for `lazyfree_objects` since we also free it in the bio thread.

sundb commented on src/lazyfree.c:
```diff
@@ -6,66 +6,55 @@
+/* lazyfree generic function is to call free function, then decrease
+ * lazyfree_objects and increase lazyfreed_objects by specific size. */
+#define lazyfreeGeneric(freeFunction, arg, size) do {  \
+    size_t original_size = size;                       \
```
nitpick, I think we should add `()` for them.
```suggestion
    size_t original_size = (size);                     \
```

ShooterIT replied:
don't think it is necessary, since it is the assignment operation, there is no bad case, i think.

madolson commented:
@ShooterIT I'm not sure this makes the code all that much more maintainable, and messes with git history. So would prefer not to merge it at this time.

ShooterIT commented:
Hi @madolson , as we know, client output buffer may be extremely big and we would free it after reaching client output buffer limit, this will make server blocking but we have no clue to find, so  i implemented `lazy free client buffer`. But when i added some codes into `lazyfree.c`, i found there are many similar codes, just free `dict`, `list`, `rax`. And i think we maybe lazy free other things which also are basic data structs  like `dict`, `list`, `rax`, so i refactored it.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/redis/redis?pullRequest=10600) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/redis/redis?pullRequest=10600) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/redis/redis?pullRequest=10600) it.</sub>
